### PR TITLE
Yaml config

### PR DESCRIPTION
**Pull request by marcelhollerbach**
The new config system!

### CODE REVIEW

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java [hunk 1]
```diff
@@ -33,49 +36,50 @@
...
-        Properties properties = new Properties();
+        Config config = Yaml.loadType(Main.class.getResourceAsStream("/config.yaml"), Config.class);
 
-        try (InputStream in = Main.class.getResourceAsStream("/config.properties")) {
-            properties.load(in);
-        } catch (Exception e) {
-            LOGGER.error("Could not read application configuration.", e);
-            System.exit(1);
-        }
+        if (config.database.maintainInterval == 0)
```
Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java:45]: <= 0?

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java [hunk 2]
```diff
@@ -34,49 +37,50 @@
         System.setProperty("org.jooq.no-logo", "true");
     }
 
-    public static void main(String[] args) throws IOException {
+    public static void main(String[] args) throws IOException, ConfigException {
         LOGGER.trace("Entering application.");
 
-        Properties properties = new Properties();
+        Config config = Yaml.loadType(Main.class.getResourceAsStream("/config.yaml"), Config.class);
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: `yaml` vs. `yml`?

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java [hunk 3]
```diff
@@ -34,49 +37,50 @@
...
 
-        try (InputStream in = Main.class.getResourceAsStream("/config.properties")) {
-            properties.load(in);
-        } catch (Exception e) {
-            LOGGER.error("Could not read application configuration.", e);
-            System.exit(1);
-        }
+        if (config.database.maintainInterval < 1)
+            config.database.maintainInterval = 24;
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: Curly braces.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java [hunk 4]
```diff
@@ -34,49 +37,50 @@
...
+                    //TODO somone needs to implement mturk ... NOOOONE
+                    throw new IllegalArgumentException("Nonono we cannot do this now.");
+                case "pybossa":
+                    //TODO someone needs to implement pybossa SIIIIMON
+                    throw new IllegalArgumentException("Nonono we cannot do this now.");
+                case "dummy":
+                    platformInstance = new DummyPlatform(platform.name);
+                    break;
+                default:
+                    throw new ConfigException("Platform type "+platform.type+" not found");
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: Maybe single quotes around the type so it's clear when it's empty.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java (hunk 3 above)
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: dont see a reason for it.

Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: Because we use them everywhere. /cc @LeanderK

Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: Invalid configuration should probably throw up or at least log a warning.

Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: but you dont need them here

**Comment by marcelhollerbach** [comment]
push!

**Comment by LeanderK** [comment]
yeah!

**Pull request merged**